### ISSUE TITLE
Route null on navigation - While using children routes

### ISSUE THREAD

**Issue by Shadowman4205** [kriasoft/react-starter-kit#1227]
I downloaded react-starter-kit intl with upgraded version and  i get error with route null. 
I was using previus version with webpack v1.And there is working well!
As i test using keyword debugger the error happen in routes with children, because 1st it goes to this code
```
code 1st
 async action({ next }) {
    // Execute each child route until one of them return the result
    const route = await next(); //null

    // Provide default values for title, description etc.
    route.title = `${route.title || 'Untitled Page'} - www.reactstarterkit.com`;
    route.description = route.description || '';

    return route;
  },
```
and after that it goes here

```
code2ond
export default {
  path: '/news',
  children: [
    {
      path: '/',
      async action({ path, store }) {
      return .....
     },
     {
      path: '*',
      async action({ path, store }) {
      return .....
     },
 ]
```
but it should be backwards. code2nd -> code1st.
Am i missing something with this new version?

**Comment by Shadowman4205** [comment]
By doing this the docker image is reduced by 30ΜΒ.
Dockerfile:
```
FROM node:7.9.0-alpine

# Copy application files
COPY ./build /usr/src/app
WORKDIR /usr/src/app

# Install Node.js dependencies
RUN yarn install --production --no-progress

CMD [ "node", "server.js" ]
```
Is there something more to do?

**Comment by frenzzy** [comment]
Try to use `next(true)` as a temporary solution

**Comment by langpavel** [comment]
@frenzzy Do you have a sketch for fixing this?

**Comment by frenzzy** [comment]
Yes, here: https://github.com/kriasoft/universal-router/pull/91

**Comment by frenzzy** [comment]
Fixed in [`universal-router v3.1.0`](https://github.com/kriasoft/universal-router/releases/tag/v3.1.0) and updated in #1237
